server/src/main/java/org/elasticsearch/search/aggregations/bucket/terms/GlobalOrdinalsStringTermsAggregator.java

```diff
@@ -104,7 +106,13 @@ public GlobalOrdinalsStringTermsAggregator(
         this.valueCount = values.getValueCount();
         this.lookupGlobalOrd = values::lookupOrd;
         this.acceptedGlobalOrdinals = includeExclude == null ? ALWAYS_TRUE : includeExclude.acceptedGlobalOrdinals(values)::get;
-        this.collectionStrategy = remapGlobalOrds ? new RemapGlobalOrds() : new DenseGlobalOrds();
+        if (remapGlobalOrds) {
+            this.collectionStrategy = new RemapGlobalOrds(collectsFromSingleBucket);
+        } else {
+            // Dense ords don't know how to collect from many buckets
+            assert collectsFromSingleBucket;
```

Comment:
How confidant are we that this assert will never trip?  I'm wondering if it makes sense to change the condition to read `if (remapGlobalOrds || collectFromSingleBucket==false)`

Reply:
I'm pretty confident. But I'll switch it to a hard check just for extra assurance.